app/services/assessor/AssessorService.scala

```diff
@@ -92,6 +84,59 @@ trait AssessorService {
     }
   }
 
+  private def updateAssessor(userId: String, assessor: model.exchange.Assessor, existing: Assessor): Future[Unit] = {
+    val oldSkills = existing.skills.toSet
+    val newSkills = assessor.skills.toSet
+    val skillsToRemove = oldSkills -- newSkills
+
+    hasFutureAssessorAllocations(userId, skillsToRemove).map { hasFutureAssessorAllocationsVal =>
+      if (hasFutureAssessorAllocationsVal) {
+        throw new CannotUpdateAssessorWhenSkillsAreRemovedAndFutureAllocationExistsException(userId,
+          s"You cannot remove skills from user with id $userId when the user has been allocated to those future events with those skills." +
+          " Please remove the allocations from those events before removing the skills")
+      } else {
+        val assessorToPersist = model.persisted.assessor.Assessor(
+          userId, newVersion, assessor.skills, assessor.sifterSchemes, assessor.civilServant, existing.availability, existing.status
+        )
+        assessorRepository.save(assessorToPersist).map(_ => ())
+      }
+    }
+  }
+
+  private def hasFutureAssessorAllocations(userId: String, skills: Set[String]): Future[Boolean] = {
+    def filterOnlyFutureAssessorAllocations(assessorAllocations: Seq[AssessorAllocation]) = {
+      def addIsFutureEvent(assessorAllocations: Seq[AssessorAllocation]) = {
+        def isFutureEvent(eventId: String): Future[Boolean] = {
+          eventsService.getEvent(eventId).map { event =>
+            val eventDate = event.date
+            val today = LocalDate.now()
```

Comment:
You create today variable for each event. Move it out -  to `hasFutureAssessorAllocations` level, so it evaluated once.